packages/core/src/state-management/__tests__/repository.test.ts

```diff
@@ -637,3 +649,105 @@ describe('applyWriteOpts', () => {
+      handleAnchorCommitSpy.mockImplementation(async (state$: RunningState) => {
+        state$.next({
+          ...state$.value,
+          anchorStatus: AnchorStatus.ANCHORED,
+        })
+      })
+      const shouldRemove = await repository.handleAnchorEvent(state$, event)
+      expect(shouldRemove).toBeTruthy()
+      expect(handleAnchorCommitSpy).toBeCalled()
+      expect(state$.state.anchorStatus).toEqual(AnchorStatus.ANCHORED)
+    })
+
+    test('Anchor FAILED for tip should update the stream state and be removed from the anchorRequestStore', async () => {
+      const tile = await TileDocument.create(ceramic, { text: 1 }, undefined, { anchor: true })
+      await tile.update({ text: 2 }, undefined, { anchor: true })
+      const state$ = await repository.load(tile.id)
+      const event: AnchorEvent = {
+        status: AnchorRequestStatusName.FAILED,
+        message: 'Last FAILED',
+        streamId: tile.id,
+        cid: tile.tip,
+      }
+      const shouldRemove = await repository.handleAnchorEvent(state$, event)
+      expect(shouldRemove).toBeTruthy()
+      expect(state$.state.anchorStatus).toEqual(AnchorStatus.FAILED)
+    })
+  })
+
+  describe('for non-tip', () => {
+    test('Anchor COMPLETED for non tip should remove the request from the store', async () => {
+      // The anchor commit gets applied anyway
+      const tile = await TileDocument.create(ceramic, { text: 1 }, undefined, { anchor: true })
+      await tile.update({ text: 2 }, undefined, { anchor: true })
+      const state$ = await repository.load(tile.id)
+      const event: AnchorEvent = {
+        status: AnchorRequestStatusName.COMPLETED,
+        message: 'First COMPLETED',
+        streamId: tile.id,
+        cid: tile.state.log[0].cid,
+        witnessCar: generateFakeCarFile(),
+      }
+      const handleAnchorCommitSpy = jest.spyOn(repository as any, '_handleAnchorCommit')
+      handleAnchorCommitSpy.mockImplementation(async (state$: RunningState) => {
+        state$.next({
+          ...state$.value,
+          anchorStatus: AnchorStatus.ANCHORED,
+        })
+      })
+      const shouldRemove = await repository.handleAnchorEvent(state$, event)
+      expect(shouldRemove).toBeTruthy()
+      expect(handleAnchorCommitSpy).toBeCalled()
+    })
+
+    test('Anchor REPLACED for non tip should remove the request from the store', async () => {
+      const tile = await TileDocument.create(ceramic, { text: 1 }, undefined, { anchor: true })
+      await tile.update({ text: 2 }, undefined, { anchor: true })
+      const state$ = await repository.load(tile.id)
+      const event: AnchorEvent = {
+        status: AnchorRequestStatusName.REPLACED,
+        message: 'First REPLACED',
+        streamId: tile.id,
+        cid: tile.state.log[0].cid,
+      }
+      const nextSpy = jest.spyOn(state$, 'next')
+      const shouldRemove = await repository.handleAnchorEvent(state$, event)
+      expect(shouldRemove).toBeTruthy()
```

Comment:
this should be false no?  We still need to keep polling for the tip, the fact that the non-tip previous commit was REPLACED doesn't change that.